src/hu/openig/scripting/missions/Mission21.java

```diff
@@ -266,10 +267,16 @@ public void onAutobattleFinish(BattleInfo battle) {
 	 */
 	boolean concludeBattle(BattleInfo battle) {
 		boolean result = false;
+		boolean prototypeSurvived = true;
 		clearMission("Mission-21-Timeout");
-		Fleet carrier = findTaggedFleet(ALLY, player);
 
-		if (carrier != null) {
+		for (SpacewarStructure s: battle.spaceLosses) {

```

Comment:
`findTaggedFleet` could locate `Mission-21-Prototype` and thus no need for looping through the losses.